functions/Get-DbaConfiguration.ps1

```diff
@@ -0,0 +1,108 @@
+﻿FUNCTION Get-DbaConfiguration
+{
+<#
+.SYNOPSIS
+Returns all server level system configuration (sys.configuration/sp_configure) information
+
+.DESCRIPTION
+This function returns server level system configuration (sys.configuration/sp_configure) information. The information is gathered through SMO Configuration.Properties and be returned either in detailed or standard format
+
+.PARAMETER SqlServer
+SQLServer name or SMO object representing the SQL Server to connect to. This can be a
+collection and recieve pipeline input
+
+.PARAMETER SqlCredential
+PSCredential object to connect as. If not specified, current Windows login will be used.
+
+.PARAMETER Detailed
+Returns more information about the configuration settings than standard
+
+.NOTES 
+Original Author: Nic Cain, https://sirsql.net/
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.	
+
+.LINK
+https://dbatools.io/Get-DbaConfiguration
+
+.EXAMPLE
+Get-DbaConfiguration -SqlServer localhost
+Returns server level configuration data on the localhost (ServerName, ConfigName, DisplayName, ConfiguredValue, CurrentlyRunningValue)
+
+.EXAMPLE
+Get-DbaConfiguration -SqlServer localhost -Detailed
+Returns detailed information on server level configuration data on the localhost (ServerName, ConfigName, DisplayName, Description, IsAdvanced, IsDynamic, MinValue, MaxValue, ConfiguredValue, CurrentlyRunningValue)
+
+.EXAMPLE
+'localhost','localhost\namedinstance' | Get-DbaConfiguration
+Returns system configuration information on multiple instances piped into the function
+
+#>
+	[CmdletBinding()]
+	Param (
+		[parameter(Position = 0, Mandatory = $true, ValueFromPipeline = $True)]
+		[Alias("ServerInstance", "SqlInstance", "SqlServers")]
+		[string[]]$SqlServer,
+		[System.Management.Automation.PSCredential]$SqlCredential,
+		[switch]$Detailed
+	)
+	BEGIN
+	{
+		$output = @()
+	}
+	PROCESS
+	{
+		FOREACH ($Server in $SqlServer)
```

Comment:
I also noticed-- this should be named differently. like foreach ($instance in $sqlserver) 


then a few lines below, $server = connect-sqlserver -sqlserver $instance